### PR TITLE
Add requirements.txt to facilitate security

### LINKED ISSUE #434: [ENH] add requirements.txt to facilitate discovery by GitHub dependency graph
## Is your feature request related to a problem? Please describe.

The dependency graph of Github doesn't show us all or dependencies, which means we wouldn't be warned if one got compromised.

## Describe the solution you'd like

Document all our dependencies as described under https://docs.github.com/en/github/visualizing-repository-data-with-graphs/about-the-dependency-graph in a `requirements.txt` so that it's discovered by the dependency graph. Point the developer docs to it in order to avoid duplicate information.

## Describe alternatives you've considered

Use `pipfile.lock` but it looks too early, too complex, without clear advantage.

## Additional context

A way to define a dependency to librsync would be nice but I don't see any option.

### PR DESCRIPTION
**Pull request by ericzolf**
DEV: add requirements.txt to help GitHub detect our dependencies and warn about security flaws, closes #434

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+setuptools-scm
+importlib-metadata ~= 1.0 ; python_version < "3.8"
+
+# optional
+pylibacl
+pyxattr
+# or xattr under SuSE Linux & Co
+
+# purely for development and testing purposes
+tox
```
Comment (ikus060) [inline requirements.txt:23]: Developement dependencies should not be listed here.

Comment (ericzolf) [reply requirements.txt:23]: Why not? The file isn't of any use beside telling us if there is some security flaw in our dependencies, and I think it's good to know also about flaws in our development tooling.

**Comment by ericzolf** [comment]
Last forced push was to resolve the merge conflict.

File: requirements.txt (hunk 1 above)
Comment (ericzolf) [reply requirements.txt:23]: I consider the point closed by lack of response, so that I can merge myself.

**Pull request merged**